cfgov/housing_counselor/enforcement_list_saver.py

```diff
@@ -0,0 +1,24 @@
+import json
+from zipfile import ZIP_DEFLATED, ZipFile
+
+
+json_file_path = './HUD_approved_housing_counselors.json'
+
+
+# By request from stakeholders in the Office of Innovation, save a
+# list of the full results we receive from the HUD API, in case of
+# future enforcement actions.
+
+# As an MVP, we're saving the raw JSON response, zipped. As a future
+# enhancement, we could consider improving the format of the data,
+# e.g. as CSV.
+def save_list(counselors, destination_path):
+    with open(json_file_path, 'w') as f:
+        f.write(json.dumps(counselors))
+
+    if not destination_path:
```

Comment:
Is there an advantage to making this argument optional, instead of making it required? It'd simplify the code to make it always required, and avoid the need to hardcode a default that won't be used in practice.

Reply:
If the call to the `hud_generate_json` management command doesn't include a value for the `--full-file-name` argument, then the `full_file_name` variable ends up having a value of `None`. In order to use the `save_list` function with an optional parameter and a default value, I'd have to call the method like this:

```python
        if full_file_name:
            save_list(counselors, full_file_name)
        else:
            save_list(counselors)
```
Is that what you're suggesting? I'm probably misunderstanding your comment.

Reply:
Is there a good reason to make the `--full-file-name` parameter optional? Adding `required=True` to its definition would make it required, preventing the script from running if it isn't provided.

If saving the full archive isn't optional -- meaning that it always gets saved regardless -- then I think it's reasonable to require an output filename.

Reply:
Added an update to give that parameter a better name and make it required. Thanks for the suggestion.